File: 53-Maximum-Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxsub = nums[0]
        cursum = 0
        for num in nums:
            cursum = max(num, cursum + num)
            maxsub = max(maxsub, cursum)
        return maxsub

File: 53-Maximum-Subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
       
      int curSum=0;
      int maxSum=nums[0];
       for(int val:nums){
        curSum+=val;
       // maxSum=Math.max(curSum,maxSum);
         maxSum = curSum> maxSum? curSum : maxSum ;
        if(curSum<0){
            curSum=0;
        }
       }
    return maxSum;
    }
}